tests/kernel/mem_protect/obj_validation/src/main.c

```diff
@@ -112,10 +121,109 @@ void test_generic_object(void)
 	}
 }
 
+/**
+ * @brief Test requestor thread will implicitly be assigned permission on the
+ * dynamically allocated object
+ *
+ * @details
+ * - Create kernel object semaphore, dynamically allocate it from the calling
+ *   thread's resource pool.
+ * - Check that object's address is in bounds of that memory pool.
+ * - Then check the requestor thread will implicitly be assigned permission on
+ *   the allocated object by using semaphore API k_sem_init()
+ *
+ * @ingroup kernel_memprotect_tests
+ *
+ * @see k_object_alloc()
+ */
+void test_kobj_assign_perms_on_alloc_obj(void)
+{
+	test_dyn_sem0 = k_object_alloc(K_OBJ_SEM);
+	zassert_not_null(&test_dyn_sem0, "Cannot allocate sem k_object");
```

Comment:
@andrewboie I emailed you about that test

Reply:
you've totally lost me

answer the question: why are you testing the ADDRESS of a pointer variable on the stack and not its contents? This is basic C coding concepts.

Reply:
I understand that I should use * to access data through the pointer which stores the address of the variable. Here I was wrong, I can't explain why exactly I wrote that line in the past.

Reply:
I made changes, I understand now that using & in that test was a mistake, because 
```
zassert_not_null(ptr, msg, ...)
Assert that ptr is not NULL.
```